### PR TITLE
add dx history finder for procedure order form

### PR DESCRIPTION
**Pull request by sjpadgett**
Kind of neat feature. 
- Accumulates  dx codes from current patient issues/problems and global past procedures for easy reuse on new procedure forms.
- Reduced form details to one table header, cleaner.
- Fixed js error for redeclare of mypcc.

### CODE REVIEW

**Comment by sjpadgett** [comment]
See issue #2750 for some details.

File: interface/patient_file/encounter/find_code_history.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                    <th><?php echo xlt('Origin'); ?></th>
+                    <th><?php echo xlt('Code'); ?></th>
+                    <th><?php echo xlt('Description'); ?></th>
+                </tr>
+                </thead>
+                <tbody>
+                <?php
+                foreach ($dxcodes as $pc) {
+                    echo "<tr>\n" .
+                        "<td>" . $pc['origin'] . "</td>\n" .
```
Comment (bradymiller) [inline interface/patient_file/encounter/find_code_history.php]: dang it. I was gonna say this needed to be escaped, but it doesn't :(

File: interface/patient_file/encounter/find_code_history.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                        "<td><button class='btn btn-xs btn-default' onclick='rtnCode(this)' value='" . attr($pc['code']) . "'>" .
+                        text(explode(':', $pc['code'])[1]) . "</button></td>\n" .
+                        "<td>" . $pc['desc'] . "</td>\n" .
```
Comment (bradymiller) [inline interface/patient_file/encounter/find_code_history.php]: yes, got one :)
This should be escaped with text()

File: interface/patient_file/encounter/find_code_history.php [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    $origin = xlt('Patient Problems');
+    $probcodes = array();
+    $bld = '';
+    $dres = sqlStatementNoLog(
+        "SELECT diagnosis as codes FROM lists " .
+        "Where activity = 1 And type = ? And pid = ?",
+        array('medical_problem', $pid)
+    );
+    while ($diag = sqlFetchArray($dres)) {
+        if (strpos($diag['codes'], 'ICD') === false) {
```
Comment (bradymiller) [inline interface/patient_file/encounter/find_code_history.php]: Is the goal here to only show ICD codes or any codes that are being used as diagnosis codes (for example, in another country that uses snomed for dx codes). If goal is to show diagnosis codes, then can use the collect_codetypes() function in custom/code_types.inc.php . Check out interface/patient_file/summary/add_edit_issue.php for example of use.

**Comment by bradymiller** [comment]
hi @sjpadgett , Review complete. Looks good. Just noted minor issues. thanks! -brady

File: interface/patient_file/encounter/find_code_history.php (hunk 3 above)
Comment (bradymiller) [reply interface/patient_file/encounter/find_code_history.php]: btw, this is collected from the settings stored in Administration->Lists->Code Types

File: interface/patient_file/encounter/find_code_history.php (hunk 2 above)
Comment (sjpadgett) [reply interface/patient_file/encounter/find_code_history.php]: lol, Hey it wouldn't be a PR from Jerry unless missing an escape!

File: interface/patient_file/encounter/find_code_history.php (hunk 3 above)
Comment (sjpadgett) [reply interface/patient_file/encounter/find_code_history.php]: yep, true. I'm too fixed on lab's. Only place I care about code type. thanks

**Comment by sjpadgett** [comment]
thanks for review. another feature i'll forget to document for a year:)

**Pull request merged**